odl/solvers/advanced/proximal_operators.py

```diff
@@ -342,26 +362,44 @@ def proximal_composition(proximal, operator, mu):
 
     Notes
     -----
-    The function cannot verify that the identity holds, the user needs to
-    verify this.
+    Given a linear operator :math:`L` with the property that for a scalar
+    :math:`\\mu`
+
+        .. math::
+
+            L^*(L(x)) = \\mu * x
+
+    and a convex function :math:`F`, the following identity holds
+
+        .. math::
+
+            \mathrm{prox}_{F \circ L}(x) = x + \\frac{1}{\\mu}
+            L^* \left( \mathrm{prox}_{\\mu F}(Lx) - Lx \\right)
+
+    This factory function implements this functionality.
+
+    There is no simple formula for more general operators.
+
+    The function cannot verify that the operator is unitary, the user needs
+    to verify this.
     """
 
-    def proximal_composition_factory(step_size):
-        """Create proximal for the dual with a given step_size
+    def proximal_composition_factory(sigma):
+        """Create proximal for the dual with a given sigma
 
         Parameters
         ----------
-        step_size : positive float
+        sigma : positive float
             Step size parameter
 
         Returns
         -------
         proximal : `Operator`
-            The proximal operator of ``prox[step_size * F * L](x)``
+            The proximal operator of ``prox[sigma * F * L](x)``
         """
         Id = IdentityOperator(operator.domain)
         Ir = IdentityOperator(operator.range)
-        prox_muf = proximal(step_size)
+        prox_muf = proximal(sigma)
         return Id + (1.0 / mu) * operator.adjoint((prox_muf - Ir) * operator)
```

Comment:
There has to be a bug here, `operator.adjoint((prox_muf - Ir) * operator)` makes no sense, it should be `operator.adjoint * ((prox_muf - Ir) * operator)` I guess.